python/gui/MaaS/cbv/DatasetApiView.py

```diff
@@ -0,0 +1,62 @@
+import asyncio
+from django.http import JsonResponse
+from wsgiref.util import FileWrapper
+from django.http.response import StreamingHttpResponse
+from .AbstractDatasetView import AbstractDatasetView
+from .DatasetFileWebsocketFilelike import DatasetFileWebsocketFilelike
+import logging
+logger = logging.getLogger("gui_log")
+
+
+class DatasetApiView(AbstractDatasetView):
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def _get_dataset_content_details(self, dataset_name: str):
+        result = asyncio.get_event_loop().run_until_complete(self.dataset_client.get_dataset_content_details(name=dataset_name))
+        logger.info(result)
+        return JsonResponse({"contents": result}, status=200)
+
+    def _delete_dataset(self, dataset_name: str) -> JsonResponse:
+        result = asyncio.get_event_loop().run_until_complete(self.dataset_client.delete_dataset(name=dataset_name))
+        return JsonResponse({"successful": result}, status=200)
+
+    def _get_datasets_json(self) -> JsonResponse:
+        serial_dataset_map = asyncio.get_event_loop().run_until_complete(self.get_datasets())
+        return JsonResponse({"datasets": serial_dataset_map}, status=200)
+
+    def _get_dataset_json(self, dataset_name: str) -> JsonResponse:
+        serial_dataset = asyncio.get_event_loop().run_until_complete(self.get_dataset(dataset_name=dataset_name))
+        return JsonResponse({"dataset": serial_dataset[dataset_name]}, status=200)
+
+    def _get_download(self, request, *args, **kwargs):
+        dataset_name = request.GET.get("dataset_name", None)
+        item_name = request.GET.get("item_name", None)
+        chunk_size = 8192
+
+        custom_filelike = DatasetFileWebsocketFilelike(self.dataset_client, dataset_name, item_name)
+
+        response = StreamingHttpResponse(
+            FileWrapper(custom_filelike, chunk_size),
+            content_type="application/octet-stream"
+        )
+        response['Content-Length'] = asyncio.get_event_loop().run_until_complete(self.dataset_client.get_item_size(dataset_name, item_name))
```

Comment:
`StreamingHttpResponse`s are a little wonky - I don't think they are compatible with the `Content-Length` header and will clash with Gunicorn. It might be better to have a websocket that emits this data and let the client build the blob.

Reply:
I don't have this part working anyway, so it can be implemented differently.  I can pull this and other such things out entirely, but I wasn't sure if parts might be useful to show the direction I was trying to go.

Reply:
That's precisely why it should be left in for now. I just wanted to drop some notes while I was looking at it. I don't think any of us has the bandwidth to make major changes at the moment.